### ISSUE TITLE
Support for Nvidia GT 1030

### ISSUE THREAD

**Issue by DevAndArtist** [kozlek/HWSensors#396]
Hi there, I think it's fair to say that some people from the CustoMac community including myself would really appreciate if someone could add support for the low budget pascal cards like the GT 1030.

My GT 1030 is from MSI and is passive cooled. Sometimes the card is getting really hot but there is no way to tell what the temperature is (pure macOS system). I only can "hope" that the card is under 80C° under stress. The latest kexts from `https://github.com/kozlek/HWSensors/tree/6.26` do not support this card, or at least the support is bugged, but I cannot tell for sure.

I already tried everything:
* /EFI/CLOVER/kexts/Other/
* /S/L/E/
* /L/E/

The other sensors load from any of these folders and work just fine, but only GPU sensors are missing.

**Comment by dmatora** [comment]
@DevAndArtist enjoy

**Comment by chris1111** [comment]
Thank you

**Comment by DevAndArtist** [comment]
@dmatora thank you, but how do I get it to work? I tried to compile the project in different ways. I placed the compiled `GPUSensors.kext` in clover, it didn't work, then in /S/L/E/ with the same result. Actually I replaced all the kext's that I had with the compiled ones and it broke my system because the `FakeSMC.kext` wasn't working as expected. I was stuck for 5 hours searching for a solution until I restored a previous version of `FakeSMC.kext` and my system finally could boot. Lesson learned `FakeSMC.kext` is mandatory and should not be corrupted. That said, I'm probably doing something wrong and unable to create working kext's.

@kozlek mind a new release for the current state of the project?

**Comment by chris1111** [comment]
The release V6 26  1440 works good here High Sierra 10.13.3

**Comment by DevAndArtist** [comment]
@chris1111 for me `HWMonitor` nor `iStat Menu` does not recognize the temperature for my MSI GT 1030. `iStat Menu` however is able to display FPS for the graphics card. I hope the fix from @dmatora will finally allow me to view the current temperature.

**Comment by chris1111** [comment]
I will test right now to my  Zotac GeForce GT 1030 2 gig and I let you know

**Comment by chris1111** [comment]
Post upper  edited

**Comment by DevAndArtist** [comment]
@chris1111 Yeah everything works great except for the temperature for that card.

**Comment by dmatora** [comment]
>Actually I replaced all the kext's that I had with the compiled ones and it broke my system because the FakeSMC.kext wasn't working as expected
@DevAndArtist looks like you've failed to install kexts properly

@chris1111 does it now show temperature for your GT 1030?

**Comment by DevAndArtist** [comment]
@dmatora mind to put a small instruction how to compile the project properly if you start by checking out the repo?!

**Comment by dmatora** [comment]
@DevAndArtist just selecting "Build kexts" on tasks and pressing play button builds the kexts under the Build subdirectory

**Comment by DevAndArtist** [comment]
@dmatora did that, but it won't be signed nor does the kext has a version tag attached to it. I fail to run create installer though. 

I also tried to select the `GPUSensors` target and enable automatic signing in Xcode 9, but this is the current version I have in /S/L/E which doesn't seem to run.

**Comment by dmatora** [comment]
you don't really need to sign it, to run it

**Comment by dmatora** [comment]
@DevAndArtist so does temperature sensor works for you now?

**Comment by DevAndArtist** [comment]
@dmatora yes I tested the card playing a game for 30 minutes it gets up to 82C°, which is fine for a passive cooled card. Without 3D tasks it's around 35-45C°. We now only need NVidia to fix their drivers. :) Thank you again.

**Comment by dmatora** [comment]
@DevAndArtist my pleasure

**Comment by dmatora** [comment]
@chris1111 you might want to update your build

**Comment by DevAndArtist** [comment]
@dmatora btw. is it normal that I now have an entry for GPU PWM? It does not make sense. It's running at 100% or 100rpm (depends on the app I'm using), but there is no fan for that.

**Comment by dmatora** [comment]
@DevAndArtist kext doesn't have dedicated support for at least that particular passive card, it just thinks all 1030 have fans.

**Comment by DevAndArtist** [comment]
@dmatora okay, I just wanted to know. I mean theare are 1030 cards with fans out there so it's fine, I'll just filter that entry out in iStat.

**Comment by chris1111** [comment]
I am not able to compilling the source with a Xcode 9.2 (9C40b) Error missing version.h

**Comment by DevAndArtist** [comment]
@chris1111 you have to use `HWSensors.xcworkspace` not `.xcodeproj`.

**Comment by chris1111** [comment]
@Adrian Z. Trying both, same error

**Comment by DevAndArtist** [comment]
@chris1111 which build scheme are you using? `Build kext` is the right one.

**Comment by chris1111** [comment]
@Adrian Z. Please post pics of the selection I am not see where

**Comment by DevAndArtist** [comment]
![bildschirmfoto 2018-03-02 um 19 25 33](https://user-images.githubusercontent.com/8276230/36915033-850c3014-1e4f-11e8-9a5c-04858c5aa4a1.png)

Then hit the play button.

![bildschirmfoto 2018-03-02 um 19 24 11](https://user-images.githubusercontent.com/8276230/36914986-61b69d52-1e4f-11e8-8be2-d85a2cc4ec68.png)

It should build just fine.

![bildschirmfoto 2018-03-02 um 19 24 31](https://user-images.githubusercontent.com/8276230/36914994-63e4ccc0-1e4f-11e8-96ac-985840454418.png)

Locate the kext in finder and proceed like I described in a post above.

**Comment by chris1111** [comment]
Got it thanks 
[![Modular Image Creation](https://i62.servimg.com/u/f62/18/50/18/69/captu358.png)

**Comment by chris1111** [comment]
I have make mistake, my binaries not build correctly 🧐

**Comment by DevAndArtist** [comment]
@chris1111 I only use `GPUSensors.kext` in /S/L/E, all other kext are from older 1440 release.

Have you set the Version and Build as shown in my screenshot? After reboot (don't forget to rebuild permissions and cache using kext-wizzard) open terminal and type `kextstat` and search for `GPUSensors`. If it's there then it's running correctly.

**Comment by chris1111** [comment]
@DevAndArtist No I use it in Clover, I ever used FakeSMC in clover Why S/L/E ?

**Comment by DevAndArtist** [comment]
It's up to you where you'd install them, both should work I guess. I think MultiBeast will install FakeSMC into /S/L/E by default so I just did that. You can always try /EFI/EFI/CLOVER/kexts/Other and see if it works.

**Comment by chris1111** [comment]
My question is Why GeforceSensor.kext and FakeSMC 3.5.1  from Slice works with my Geforce GT 1030 Since long time and here I have trouble ?

**Comment by DevAndArtist** [comment]
@chris1111 no idea sorry.

**Comment by dmatora** [comment]
@chris1111 there are much more people involved into developing slice fork

**Comment by chris1111** [comment]
Thank you anyway mr @DevAndArtist and  @dmatora  😇

**Comment by dmatora** [comment]
>I have make mistake, my binaries not build correctly 🧐 system not bootable
I don't think you can build wrong kexts, they are either done building or not
system not bootable most definitely means you didn't install kexts properly

**Comment by chris1111** [comment]
Kext is in Clover and the build looks good but system stall very shortly

**Comment by DevAndArtist** [comment]
If it's `AppleACPICPU` thing then your `FakeSMC` is broken, just use the one from 6.26.1440 binary build.

**Comment by chris1111** [comment]
For now I restart the system with a rescue SSD, thats not a problem for me Its hang at ACPI some thing like this very fast, I know its in relation of my build, Fakesmc must not have built correctly

**Comment by DevAndArtist** [comment]
Replace your compiled FakeSMC with the one from here https://github.com/kozlek/HWSensors/releases/download/6.26/HWSensors.6.26.1440.Binaries.dmg.zip and that's it. 

I use all kext from this package *except* for the `GPUSensors` which is the one we compiled previously.

For me everything is in S/L/E, but in CLOVER it should also be fine.

If you added, deleted or edited any kext in /S/L/E then you MUST rebuild permissions and restore caches.

**Comment by dmatora** [comment]
@chris1111 I've sent build 1462 to your email

**Comment by chris1111** [comment]
@dmatora  I revive a Email and .dmg I test the file right now thank you

**Comment by chris1111** [comment]
Its work !! 😎 Thank you mr @dmatora 
![Modular Image Creation](https://i62.servimg.com/u/f62/18/50/18/69/sans_t45.jpg)

**Comment by dmatora** [comment]
@chris1111 you're welcome

**Comment by chris1111** [comment]
@dmatora do you have the same version of HWMonitor.app please

**Comment by chris1111** [comment]
Ok got it I build successfully the same Build
 ![Modular Image Creation](https://i62.servimg.com/u/f62/18/50/18/69/captu359.png)

**Comment by chris1111** [comment]
Hi @DevAndArtist  I resolved the compile issue  
my mistake is I forgot to put a bundle Identifier 
[![Modular Image Creation](https://i62.servimg.com/u/f62/18/50/18/69/15510.png)

and the other kext is 
org.hwsensors.driver.ACPISensors
org.hwsensors.driver.CPUSensors
org.hwsensors.driver.GPUSensors
org.hwsensors.driver.LPCSensors

all kext is compile great and its working like a charm, thanks both of you @DevAndArtist , @dmatora

**Comment by ghost** [comment]
I was unable to boot with my build version of the kext's. Can anyone share their build last version of the kexts? 
Thanks in advance!

**Comment by DevAndArtist** [comment]
@emiliogonzalezjunior have you rebuild your caches and restored permissions after messing around with /S/L/E (assuming your kexts are there)?

**Comment by ghost** [comment]
I use the EFI/ other folder. I never installed anything in S/L/E

**Comment by chris1111** [comment]
@emiliogonzalezjunior you can try this Package, build  6.26. 1464
https://github.com/chris1111/HWSensors-V6.26.1464-EFI

Also if you preferred only binaries Attaching here Build 6.26.1464.zip
[Build 6.26.1464.zip](https://github.com/kozlek/HWSensors/files/1853037/Build.6.26.1464.zip)

**Comment by ghost** [comment]
Thank you very much @chris1111 !
